### PR TITLE
Feature/plcn 310 create location filter

### PR DESCRIPTION
**Pull request by s-bogelund**
LocationFilter component has been added but still lacks gql integration.
The main component is the generic CheckboxGroup component.
As of now, it just uses dummyData in its Wall of Clients implementation but this should be pretty easy to spot and thus remove when integration with gql has been implemented.

### CODE REVIEW

File: src/components/common/filters/LocationFilter.tsx [hunk 1]
```diff
@@ -1,23 +1,120 @@
...
+		onFilterUpdate(checkboxState);
+	}, [checkboxState, onFilterUpdate]);
+
+	const anyThingChecked = useMemo(() => {
+		return checkboxState.some(checkbox => checkbox.checked);
+	}, [checkboxState]);
+
+	const allChecked = useMemo(() => {
+		return checkboxState.every(checkbox => checkbox.checked);
+	}, [checkboxState]);
```
Comment (Nicolai-Bak) [inline src/components/common/filters/LocationFilter.tsx:41]: Would name the two functions above `isAnythingChecked` and `isAllChecked` to indicate a boolean is returned

File: src/components/common/filters/LocationFilter.tsx [hunk 2]
```diff
@@ -1,23 +1,120 @@
...
+ */
+const LocationFilter: FC<LocationFilterProps> = ({ locations, onFilterUpdate }) => {
+	const [checkboxState, setCheckboxState] = useState([] as CheckboxInfo[]);
+
+	useMemo(() => {
+		const initialState = locations.map(location => {
+			return {
+				label: location,
+				name: location,
+				checked: false,
```
Comment (Nicolai-Bak) [inline src/components/common/filters/LocationFilter.tsx:24]: Would name the field `isChecked` to indicate it is a boolean, but I'm not sure if you use this convention in the rest of the project.

File: src/components/common/filters/PopupFilterWrapper.tsx [hunk 3]
```diff
@@ -52,7 +52,15 @@ const PopupFilterWrapper: FC<PopupFilterWrapperProps> = ({
 			>
 				<AnimatePresence>{active && <ClearFilter onClick={onClearFilter} />}</AnimatePresence>
 				<Box>
-					<Typography ml={active ? '10px' : 0} variant="body">
+					<Typography
+						ml={active ? '10px' : 0}
```
Comment (itm-mne) [inline src/components/common/filters/PopupFilterWrapper.tsx:56]: Could consider using `theme.spacing`instead.

**Review by itm-mne — APPROVED** [review]
Only some minor suggestions, otherwise looking fine.

**Pull request merged**